### PR TITLE
RHIROS-783 fix db query for /call_to_action

### PR DESCRIPTION
**Pull request by r14chandra**

### CODE REVIEW

File: ros/api/v1/call_to_action.py [hunk 1]
```diff
@@ -1,9 +1,9 @@
 from flask_restful import Resource
 from flask import request
 from ros.lib.utils import (
-    org_id_from_identity_header, system_ids_by_org_id)
+    org_id_from_identity_header, systems_with_profile)
```
Comment (kgaikwad) [inline ros/api/v1/call_to_action.py:4]: @r14chandra - Any specific reason that you extracted query part into a separate method?

Comment (r14chandra) [reply ros/api/v1/call_to_action.py:4]: We can use this method in the PR for /is_configured - #254. Once this PR is in, I'll make the required changes in that PR.

Comment (r14chandra) [reply ros/api/v1/call_to_action.py:4]: No. of systems having performance profile is used in some place like under `is_configured` and `executive_report`

File: ros/lib/utils.py [hunk 2]
```diff
@@ -158,6 +158,11 @@ def calculate_percentage(numerator, denominator):
         return 0
 
 
+def systems_with_profile(org_id):
```
Comment (kgaikwad) [inline ros/lib/utils.py:161]: Will it make sense to include one more function argument which will accept the extra filters specific to context?

Comment (kgaikwad) [inline ros/lib/utils.py:161]:
```suggestion
def systems_ids_with_profile(org_id):
```

This method returns system ids and not system/profile records. Could you rename this method to something more specific?

**Review by kgaikwad — CHANGES_REQUESTED** [review]

File: ros/lib/utils.py (hunk 2 above)
Comment (r14chandra) [reply ros/lib/utils.py:161]: I've only seen filter for `number_of_recommendations`. Not generic. Is it still needed?

Comment (kgaikwad) [reply ros/lib/utils.py]: > I've only seen filter for `number_of_recommendations`. Not generic. Is it still needed?

Then, it won't be needed. One query - do we really required `number_of_recommendations` filter? In which scenario system will have a performance profile record without any recommendation?

Comment (r14chandra) [reply ros/lib/utils.py]: `Optimized` state archive.

**Review by kgaikwad — APPROVED** [review]
LGTM 👍

**Comment by kgaikwad** [comment]
Thanks @r14chandra!

**Pull request merged**